nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/CreateOrUpdateGroup.java

```diff
@@ -168,4 +188,37 @@ public static String getTenantGroupName(String groupName, String tenantId) {
         return BaseSession.computeMultiTenantDirectoryId(tenantId, groupName);
     }
 
+    protected void checkCanCreateOrUpdateGroup(DocumentModel groupDoc) {
+        NuxeoPrincipal currentUser = ctx.getPrincipal();
+        if (!currentUser.isAdministrator()
+                && (!currentUser.isMemberOf("powerusers") || !canCreateOrUpdateGroup(groupDoc))) {
+            throw new NuxeoException("User is not allowed to create or edit groups", HttpServletResponse.SC_FORBIDDEN);
+        }
+    }
+
+    protected boolean canCreateOrUpdateGroup(DocumentModel groupDoc) {
+        GroupConfig groupConfig = userManager.getGroupConfig();
+        NuxeoGroup group = new NuxeoGroupImpl(groupDoc, groupConfig);
+        List<String> allGroups = computeAllGroups(group);
+        List<String> administratorsGroups = userManager.getAdministratorsGroups();
+        return allGroups.stream().noneMatch(administratorsGroups::contains);
+    }
+
+    protected List<String> computeAllGroups(NuxeoGroup group) {
+        List<String> allGroups = new ArrayList<>();
+        computeAllGroups(allGroups, group);
+        return allGroups;
+    }
+
+    protected void computeAllGroups(List<String> allGroups, NuxeoGroup group) {
+        allGroups.add(group.getName());
+        group.getParentGroups().stream().filter(pg -> !allGroups.contains(pg)).forEach(pg -> {
+            allGroups.add(pg);
+            NuxeoGroup parentGroup = userManager.getGroup(pg);
+            if (parentGroup != null) {
+                computeAllGroups(allGroups, parentGroup);
+            }
+        });
+    }
```

Comment:
i don't remember but shouldn't we add the javadoc on the new API (`public` method i am sure but what about `protected`)

Reply:
Not strictly needed.